### PR TITLE
OOP school library: Add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by Rudy-Menekam**
1. Create class Person, student, and teacher
2. Implement encapsulation and inheritance with Ruby.

### CODE REVIEW

File: teacher.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+class Teacher
+  def initialize(specialization, age, name = 'unknown', parent_permission: true)
+    super(age, name, parent_permission)
+    @specialization = specialization
+  end
+
+  def can_use_services?
+    true
+  end
+end
```
Comment (mckent05) [inline teacher.rb]: - According to the project requirements, please ensure that your `Teacher` class inherits from the `Person` class. This [article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_principles_examples.md#inheritance) will help you understand better. Please fix this 🙏

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+class Student
+  def initialize(classroom, age, name = 'unknown', parent_permission: true)
+    super(name, age, parent_permission: parent_permission)
+    @classroom = classroom
+  end
+
+  def play_hooky
+    '¯\(ツ)/¯'
+  end
+end
```
Comment (mckent05) [inline student.rb]: - According to the project requirements, please ensure that your `Student` class inherits from the `Person` class. This [article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_principles_examples.md#inheritance) will help you understand better. Please fix this 🙏

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @Rudy-Menekam, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job implementing classes to represent students and teachers 👏 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mukhammadrizooff — APPROVED** [review]
Hi @Rudy-Menekam, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**